eng/icu.wasi.mk

```diff
@@ -0,0 +1,10 @@
+ifeq ($(WASM_ENABLE_THREADS),true)
```

Comment:
set CC to `$(WASI_SDK_PATH)/bin/clang.exe `
- i'm not sure what's the syntax here
- see https://github.com/dotnet/icu/blob/cb39ef7979da97c65ee27a626eb83b0290216298/icu/icu4c/source/configure#L2911
- possibly you need to set other flags which the configure is looking for
- learning from emscripten's approach may help
```
ENV_INIT_SCRIPT = source $(EMSDK_PATH)/emsdk_env.sh &&
ENV_CONFIGURE_WRAPPER = emconfigure
```